addons/hr_expense/models/hr_expense.py

```diff
@@ -868,14 +868,14 @@ def action_sheet_move_create(self):
         expense_line_ids = self.mapped('expense_line_ids')\
             .filtered(lambda r: not float_is_zero(r.total_amount, precision_rounding=(r.currency_id or self.env.company.currency_id).rounding))
         res = expense_line_ids.action_move_create()
+        for sheet in self:
+            if sheet.accounting_date:
+                sheet.accounting_date = sheet.account_move_id.date
```

Comment:
With the `filtered()`  you can get rid of the `if`, and thus only get the records you are interested in :-)

```suggestion
        for sheet in self.filtered(lambda s: not s.accounting_date):
            sheet.accounting_date = sheet.account_move_id.date
```